### PR TITLE
Documentation for 4.3 release

### PR DESCRIPTION
**Pull request by takluyver**
Update the changelog, document the `NotebookNotary` and new `SignatureStore` classes.

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -4,6 +4,19 @@
 Changes in nbformat
 =========================
 
+4.3
+===
+
+`4.3 on GitHub <https://github.com/jupyter/nbformat/milestone/7>`__
+
+- A new pluggable ``SignatureStore`` class allows specifying different ways to
+  record the signatures of trusted notebooks. The default is still an SQLite
```
Comment (blink1073) [inline docs/changelog.rst]: a SQLite*

Comment (takluyver) [inline docs/changelog.rst]: Interesting. I pronounce it 'ess-queue-ell-ite', so to me 'an' fits better. How do you pronounce it?

Comment (blink1073) [inline docs/changelog.rst]: Huh, TIL.  I had been using see-quill-lite

Comment (rgbkrk) [inline docs/changelog.rst]: I say "See, [Qull](http://dc.wikia.com/wiki/Qull_(New_Earth)) lite"

**Comment by takluyver** [comment]
I've added a commit to this to document the functions for constructing notebooks (`new_notebook`, `new_code_cell`, ...).

File: docs/changelog.rst (hunk 1 above)
Comment (minrk) [inline docs/changelog.rst]: My understanding is that "ess queue ell" is official (for SQL at least), but "sequel" is quite common as well. For completeness, I say "ess queue light", so we've got all the bases covered.

**Pull request merged**